### PR TITLE
Add configurable UTC timezone option for scheduler

### PR DESCRIPTION
**Pull request by pmahindrakar-oss**
Signed-off-by: pmahindrakar-oss <prafulla.mahindrakar@gmail.com>

# TL;DR

This maintain backward compat with existing scheduler deployment jobs and keeps them running in the local timezone.
Inorder to run the jobs in UTC timezone add this useUTCTz bool option in scheduler config

## Type
 - [ ] Bug Fix
 - [X] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/3047

## Follow-up issue
_NA_

### CODE REVIEW

File: pkg/runtime/interfaces/application_configuration.go [hunk 1]
```diff
@@ -344,12 +344,17 @@ type FlyteWorkflowExecutorConfig struct {
 	// eg : 100 TPS will send at the max 100 schedule requests to admin per sec.
 	// Burst specifies burst traffic count
 	AdminRateLimit *AdminRateLimit `json:"adminRateLimit"`
+	UseUTCTz       bool            `json:"useUTCTz"`
```
[the remark this answers is not in the document]
Comment (pmahindrakar-oss) [reply pkg/runtime/interfaces/application_configuration.go:347]: We can but IMO might cause difficulty in debugging scheduler issues.

All our logs are written in UTC time zone and for easier debugging having the schedule also using utc would be preferable .

Our previous aws based flyte workflow scheduler also  used UTC

https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/ScheduledEvents.html

Would prefer to keep it simple and even thinking to keep UTC as default in one of the releases

[the remark this answers is not in the document]
Comment (katrogan) [reply pkg/runtime/interfaces/application_configuration.go:347]: Sure, can we add a comment here that the default behavior is to use the user-local time zone then?

**Review by katrogan — DISMISSED** [review]

File: pkg/runtime/interfaces/application_configuration.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (pmahindrakar-oss) [reply pkg/runtime/interfaces/application_configuration.go]: Done

**Review by katrogan — APPROVED** [review]
